openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/CEEnvironmentProcessor.java

```diff
@@ -93,6 +98,11 @@ public void createEnvironment(@Observes(precedence = 10) BeforeClass event, Open
         CubeOpenShiftConfiguration configuration) {
         final TestClass testClass = event.getTestClass();
         log.info(String.format("Creating environment for %s", testClass.getName()));
+
+        if (configuration.isNamespaceClassScopeEnabled()) {
+            namespaceService.get().create(configuration.getNamespace() + "-" + testClass.getJavaClass().hashCode());
```

Comment:
With this it's quite difficult for user to debug which namaspace is created for which TestClass if he has to debug it. Can't we make it easy such a that we can understand from created namespace for which class it is created?

Reply:
The namespace can only have lowercase alphanumeric characters or `-` and should be less than 66 characters. 
I'll update if you can suggest something, having the name there doesn't work in spite of converting it to lowercase and replacing all the other characters such as `.`  etc with `-` because the length exceeds the maximum character limit. 
So, this was the only way, I could think of creating unique namespaces.

Reply:
One way for class could be `testClass.getJavaClass().getSimpleName().toLowerCase()` but then I am not sure of something equivalent for test method.

Reply:
ok as we are not sure about TestClass length, then we can show  
provide log message for it. e.g.
`Created namespace `itest-5454237` for class org.arquillian.cube.FooTest`
`Created namespace  `itest-5454267` for class:  org.arquillian.cube.FooTest method: should_show_hello_world`

Reply:
> One way for class could be testClass.getJavaClass().getSimpleName().toLowerCase() but then I am not sure of something equivalent for test method.

Not on this level, as this handles class events, but for `Before event` you get the test method easily and then `testMethod.getName()` is enough.

Reply:
I believe we can always try to use this pattern and check if we are not exceeding the constraint of 63 characters. In such a case we can rollback to something else and log as @dipak-pawar suggested.